sound/soc/soc-topology.c

```diff
@@ -1143,6 +1159,12 @@ static int soc_tplg_dapm_graph_elems_load(struct soc_tplg *tplg,
 		hdr->index);
 
 	for (i = 0; i < count; i++) {
+		struct snd_soc_dapm_route *route;
+
+		route = kzalloc(sizeof(*route), GFP_KERNEL);
+		if (!route)
+			return -ENOMEM;
```

Comment:
@plbossart Thanks for pointing this out. I need to add a sticky note on my monitor to remind myself every time I do this :)

Reply:
@plbossart I am not allocating memory twice. "routes" is a pointer to an array of pointers. And routes[i] is a pointer to snd_soc_dapm_route element. 

The reason I dont allocate them all in one go is because we delete them individually in remove_route().